### ISSUE TITLE
Static.hs build fail in sandbox, ghc 7.8.3

### ISSUE THREAD

**Issue by ocramz** [albertoruiz/hmatrix#109]
Installing the latest versions (I tried a few of the 0.16.*) by cabal install'ing (Hackage source) gives the following: it looks like a horrible low level bug, or am I missing something?

[33 of 33] Compiling Numeric.LinearAlgebra.Static ( src/Numeric/LinearAlgebra/Static.hs, dist/dist-sandbox-4b48e1be/build/Numeric/LinearAlgebra/Static.o )

/tmp/ghc79326_0/ghc79326_1.s:1:14:
     error: single quote way too long
; ModuleID = 'src/C/lapack-aux.c'
             ^

/tmp/ghc79326_0/ghc79326_1.s:1:33:
     error: single quote way too long
; ModuleID = 'src/C/lapack-aux.c'
                                ^

/tmp/ghc79326_0/ghc79326_1.s:3:15:
     error: unexpected token in argument list
target triple = "x86_64-apple-macosx10.9.0"
              ^
... and so on for a long time.

**Comment by ocramz** [comment]
this is my compiler info:
$ ghc --info
 [("Project name","The Glorious Glasgow Haskell Compilation System")
 ,("GCC extra via C opts"," -fwrapv")
 ,("C compiler command","/usr/bin/gcc")
 ,("C compiler flags"," -m64 -fno-stack-protector")
 ,("C compiler link flags"," -m64")
 ,("Haskell CPP command","/usr/bin/gcc")
 ,("Haskell CPP flags","-E -undef -traditional -Wno-invalid-pp-token -Wno-unicode -Wno-trigraphs")
 ,("ld command","/usr/bin/ld")
 ,("ld flags"," -arch x86_64")
 ,("ld supports compact unwind","YES")
 ,("ld supports build-id","NO")
 ,("ld supports filelist","YES")
 ,("ld is GNU ld","NO")
 ,("ar command","/usr/bin/ar")
 ,("ar flags","clqs")
 ,("ar supports at file","NO")
 ,("touch command","touch")
 ,("dllwrap command","/bin/false")
 ,("windres command","/bin/false")
 ,("libtool command","libtool")
 ,("perl command","/usr/bin/perl")
 ,("target os","OSDarwin")
 ,("target arch","ArchX86_64")
 ,("target word size","8")
 ,("target has GNU nonexec stack","False")
 ,("target has .ident directive","True")
 ,("target has subsections via symbols","True")
 ,("Unregisterised","NO")
 ,("LLVM llc command","llc")
 ,("LLVM opt command","opt")
 ,("Project version","7.8.3")
 ,("Booter version","7.6.3")
 ,("Stage","2")
 ,("Build platform","x86_64-apple-darwin")
 ,("Host platform","x86_64-apple-darwin")
 ,("Target platform","x86_64-apple-darwin")
 ,("Have interpreter","YES")
 ,("Object splitting supported","YES")
 ,("Have native code generator","YES")
 ,("Support SMP","YES")
 ,("Tables next to code","YES")
 ,("RTS ways","l debug thr thr_debug thr_l thr_p dyn debug_dyn thr_dyn thr_debug_dyn l_dyn thr_l_dyn")
 ,("Support dynamic-too","YES")
 ,("Support parallel --make","YES")
 ,("Dynamic by default","NO")
 ,("GHC Dynamic","YES")
 ,("Leading underscore","YES")
 ,("Debug on","False")
 ,("LibDir","/Library/Frameworks/GHC.framework/Versions/7.8.3-x86_64/usr/lib/ghc-7.8.3")
 ,("Global Package DB","/Library/Frameworks/GHC.framework/Versions/7.8.3-x86_64/usr/lib/ghc-7.8.3/package.conf.d")
 ]

**Comment by albertoruiz** [comment]
This is strange, I have never seen this kind of error. I tested the package with 7.8.4 on x86_64 (ubuntu 14.04) without any problems. I have just also tested hmatrix-0.16.1.4 in a sandbox. I don't know how to debug this.

**Comment by ocramz** [comment]
yesterday cartazio sorted me out: it's apparently a wrong default on a flag (he also filed a pull request). I managed to install correctly only by explicitly mentioning it:
`cabal install hmatrix -f-openblas`

**Comment by ocramz** [comment]
No, I confirm, installing within a sandbox, even with `f-openblas` does not work. I could only install it globally.
